rfcs/experimental/flow_graph_node_sets/README.md

```diff
@@ -0,0 +1,256 @@
+
+```cpp
+  int main()
+  {
+      tbb::flow::graph g;
+
+      tbb::flow::function_node<int,int> n2{ g, 1,
+          [](int msg) {
+              std::printf("2:%d\n", msg);
+              return msg;
+          } };
+
+      tbb::flow::function_node<int,int> n3(g, 1,
+          [](int msg) {
+              std::printf("3:%d\n", msg);
+              return msg;
+          });
+
+      using mfn_t = tbb::flow::multifunction_node<int, std::tuple<int, int> >;
+      mfn_t n1(tbb::flow::precedes(n2, n3), 1,
+          [](int msg, mfn_t::output_ports_type& p) {
+              std::printf("1:%d\n", msg);
+              std::get<0>(p).try_put(msg * 2);
+              std::get<1>(p).try_put(msg * 4);
+          });
+
+      n1.try_put(100);
+      g.wait_for_all();
+      return 0;
+  }
+```
+
+When run, the above code results in:
+
+```
+  1:100
+  3:400
+  2:200
+```
+
+The multi-port nodes include `multifunction_node`, `join_node`, `split_node` and `indexer_node`.
+The rule for creating edges from sets is the same for each of these types.
+
+### `follows` and `precedes` Helper Functions
+
+These functions creates or annotates a `node_set` with its ordering type. That is, the
+output is a node set that is a precedes or follows set. The signatures of the `follows`
+and `precedes` functions are shown below:
+
+```cpp
+    // node_set is an exposition-only name for the type returned from make_node_set function
+
+    template <typename NodeType, typename... NodeTypes>
+    /*unspecified*/ follows( node_set<NodeType, NodeTypes...>& set );  // [1]
+
+    template <typename NodeType, typename... NodeTypes>
+    /*unspecified*/ follows( NodeType& node, NodeTypes&... nodes );    // [2]
+
+    template <typename NodeType, typename... NodeTypes>
+    /*unspecified*/ precedes( node_set<NodeType, NodeTypes...>& set ); // [3]
+
+    template <typename NodeType, typename... NodeTypes>
+    /*unspecified*/ precedes( NodeType& node, NodeTypes&... nodes );   // [4]
+```
+
+The template functions `[1]` and `[3]` take a node set and add the ordering to it. For example:
+
+```cpp
+  auto handlers = make_node_set(n1, n2, n3);
+  broadcast_node<int> input(precedes(handlers));
+```
+
+While template functions `[2]` and `[4]` take a sequence of nodes and creates a node set that includes
+an ordering.  The following code is equivalent to the previous example:
+
+```cpp
+  broadcast_node<int> input(precedes(n1, n2, n3));
+```
+
+## Exit Criteria
+
+The following conditions need to be met to move the feature from experimental to fully supported:
+* Collecting feedback on user experience confirming the choices made on the open questions and limitations:
+  * Limitation that constructors can be used to set predecessors or successors but not both.
+  * The multiport node rule that makes edges from each node in the set to the corresponding port.
+* The corresponding oneTBB specification update should be done backed by the user feedback provided.
```

Comment:
With regard to the specification update, I would explicitly ask the following questions:
- Should we keep the "node set" type unspecified or define it properly?
- Should the ordering (preceding/following) become the property of a node set?

Reply:
Will add.

Reply:
Added